### DISCUSSION TITLE
Firebird connection not possible

### DISCUSSION THREAD

**Discussion by CTSchorsch** [apache/superset#29290] [Q&A / Help]
Hi, 
im quite new to superset, but it seems to be the perfect tool for us. I have a docker composer instance on my local computer up and running. i see the example dashboards. (using the official 4.01. docker image)

Now i want to add our firebird database:
- i created a docker/requirements_local.txt and add fdb and sqlalchemy-firebird to it
- i can choose firebird in the add new database dialog
- i use this as connection string -> "firebird+fdb://myuser:mypass@ip.to.db.server:3050/d:/path/to/dbfile.fdb"

i got this error message:
(builtins.NoneType) None [SQL: The location of Firebird Client Library could not be determined.] (Background on this error at: https://sqlalche.me/e/14/dbapi)

I searched through the issues and ask mr. google, but i found no hint or solution.

EDIT: It seems that the libfbclient.so is missing, but how can i add this to the image ?

Any help is appreciated
Thanks in advance
Georg

**Comment by rondinellewalitte** [comment]
Hi @CTSchorsch ,

You are right — this error usually happens because the Firebird client library is missing inside the Docker container.

To solve this, you need to install the package libfbclient2 (or the appropriate Firebird client library for your Linux distribution) inside your Superset Docker image.

If you're using a Debian-based image, you can add this to your Dockerfile:

RUN apt update && apt install -y libfbclient2

After that, rebuild your image and try the connection again.

Let me know if you need help updating your Dockerfile or rebuilding your containers.
